### PR TITLE
Patch/surprise target

### PR DESCRIPTION
**Pull request by sternhenri**
**In this PR**:
- placeholder for using a target with SurprisePoSt (TBD in research)
- moved surprise PoSt verif to the actor > subsystem

**Needed in future PR**:
- Randomness from RT
- Pulling miner actor address easily
- Getting worker public keys from miner actor address as a facility in VM

### CODE REVIEW

File: src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go [hunk 1]
```diff
@@ -381,6 +369,103 @@ func (a *StorageMinerActorCode_I) ProcessVerifiedElectionPoSt(rt Runtime) InvocO
...
+	// otherwise, abort ProcessVerifiedSurprisePoSt
+	a._ensurePledgeCollateralSatisfied(rt)
+
+	return a._onSuccessfulPoSt(rt)
+
+}
+
+func (a *StorageMinerActorCode_I) GetWorkerKeyByMinerAddress(minerAddr addr.Address) filcrypto.VRFPublicKey {
+	panic("TODO")
+}
```
Comment (anorth) [inline src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go]: This doesn't make sense here, and it looks like you removed calls to it.

What would make sense is
```
func (a *SMAC_I) GetWorkerKey() {
    h, st := a.State(rt)
    w := st.Info().Worker
    Release(...)
    return w
}
```

File: src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go [hunk 2]
```diff
@@ -381,6 +369,103 @@ func (a *StorageMinerActorCode_I) ProcessVerifiedElectionPoSt(rt Runtime) InvocO
...
+	randomness := rt.Randomness(randomnessEpoch, SPC_LOOKBACK_POST)
+	panic(randomness)                       // ignore circular dependency
+	var postRandomnessInput util.Randomness // sms.PreparePoStChallengeSeed(randomness, actorAddr)
+
+	postRand := &filcrypto.VRFResult_I{
+		Output_: onChainInfo.Randomness(),
+	}
+
+	// TODO: get worker key from minerAddr
+	var workerKey filcrypto.VRFPublicKey
```
Comment (anorth) [inline src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go]: See above, this is easy.

File: src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go [hunk 3]
```diff
@@ -381,6 +369,103 @@ func (a *StorageMinerActorCode_I) ProcessVerifiedElectionPoSt(rt Runtime) InvocO
...
+
+	// verify the partialTickets themselves
+	// 4. Verify appropriate randomness
+
+	// pull from consts
+	SPC_LOOKBACK_POST := uint64(0)
+	randomnessEpoch := st.ChallengeStatus().LastChallengeEpoch()
+	randomness := rt.Randomness(randomnessEpoch, SPC_LOOKBACK_POST)
+	panic(randomness)                       // ignore circular dependency
+	var postRandomnessInput util.Randomness // sms.PreparePoStChallengeSeed(randomness, actorAddr)
```
Comment (anorth) [inline src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go]: I don't quite understand what's going on here, what's in/complete and whether it should be straightforward in implementation.

File: src/systems/filecoin_mining/storage_mining/storage_mining_subsystem.go [hunk 4]
```diff
@@ -220,21 +216,29 @@ func (sms *StorageMiningSubsystem_I) VerifyElectionPoSt(header block.BlockHeader
...
-	postRandomnessInput := sector.PoStRandomness(sms._preparePoStChallengeSeed(randomness, header.Miner()))
+	postRandomnessInput := sector.PoStRandomness(sms.PreparePoStChallengeSeed(randomness, header.Miner()))
 
 	postRand := &filcrypto.VRFResult_I{
 		Output_: onChainInfo.Randomness(),
 	}
 
-	if !postRand.Verify(postRandomnessInput, sms.GetWorkerKeyByMinerAddress(header.Miner())) {
+	// TODO: get worker key from minerAddr
+	var workerKey filcrypto.VRFPublicKey
```
Comment (anorth) [inline src/systems/filecoin_mining/storage_mining/storage_mining_subsystem.go]: This is straightforward when you have `sma`, though you might need to add an accessor method.

**Comment by anorth** [comment]
> Getting worker public keys from miner actor address as a facility in VM

I don't think we should do this. Send a message to the actor instead.

File: src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go (hunk 3 above)
Comment (sternhenri) [reply src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go]: it should be but I'll use this to clean up randomness in the vm

**Pull request merged**